### DISCUSSION TITLE
Limitation: Background Data Sync on iOS and Android

### DISCUSSION THREAD

**Discussion by floriancb** [cofu-app/cbl-dart#530] [Q&A]
I have been wondering whether there are any fundamental reasons which would not allow cbl-dart to sync data in the background. Was there just no need to get this implemented, or is there more to it?

Any feedback is much appreciated.

**Comment by blaugold** [comment]
This limitation is more of a missing feature than a limitation, when compared to the iOS SDK. Currently, the Dart SDK does not implement a similar integration with the app life cycle [described here](https://docs.couchbase.com/couchbase-lite/current/swift/replication.html#replication-status-and-app-life-cycle). There is nothing preventing you from running a replicator while the app is in the background.

**Reply by floriancb** [reply]
Thank you, @blaugold. For clarification purposes: assume we develop a CB Lite app using cbl-dart and run it on an iOS device. Am I understanding correctly that replications (continuous, for that matter) will continue to run (and replicate) in the background?

**Reply by blaugold** [reply]
Unless the app requests more time, it has 5 more seconds of execution time after it has been put into the background. During those 5 seconds, it will continue to replicate until the app's execution is suspended. I expect that once the OS starts execution of the app again, the replicator continues to replicate, though I haven't tested that scenario. It might be that network connections have to be reestablished, and I don't know how CBL would handle this case. I'll try and test this unless you beat me to it.
